core/market/decentralized/src/negotiation/common.rs

```diff
@@ -117,15 +103,32 @@ impl CommonBroker {
 
     pub async fn get_proposal(
         &self,
-        proposal_id: &ProposalId,
-    ) -> Result<Proposal, GetProposalError> {
+        subscription_id: Option<SubscriptionId>,
+        id: &ProposalId,
+    ) -> Result<Proposal, ProposalError> {
         Ok(self
             .db
             .as_dao::<ProposalDao>()
-            .get_proposal(&proposal_id)
+            .get_proposal(&id)
             .await
-            .map_err(|e| GetProposalError::FailedGetFromDb(proposal_id.clone(), e))?
-            .ok_or_else(|| GetProposalError::NotFound(proposal_id.clone()))?)
+            .map_err(|e| ProposalError::Get(id.clone(), subscription_id.clone(), e))?
+            .filter(|proposal| {
+                if subscription_id.is_none() {
```

Comment:
Consider using `match` construct. It could probably look nicer here.

Reply:
I don't see any improvement here using `match`